h/search/query.py

```diff
@@ -165,6 +165,33 @@ def __call__(self, search, _):
         return search.filter("terms", group=groups)
 
 
+class UriWildcardFilter(object):
+
+    """
+    A filter that selects only annotations where the 'site' parameter matches as *site*.
+    """
+
+    def __init__(self, request):
+        """Searches for wildcard uri's.
+
+        :param request: the pyramid.request object
+
+        """
+        self.request = request
+
+    def __call__(self, search, params):
+        query_uris = params.getall('wildcard_uri')
+
+        uris = set()
+        for query_uri in query_uris:
+            expanded = storage.expand_uri(self.request.db, query_uri)
+
+            us = [uri.normalize(u) for u in expanded]
+            uris.update(us)
+
+        return search.query(Q('bool', should=[Q('wildcard', **{"target.scope": u}) for u in uris]))
```

Comment:
The documentation for [Wildcard Query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wildcard-query.html) notes that:

> Note that this query can be slow, as it needs to iterate over many terms. In order to prevent extremely slow wildcard queries, a wildcard term should not start with one of the wildcards * or ?.

I think we probably ought to enforce this restriction. Perhaps require that at least protocol + domain appear before the wildcard?